packages/api-db-mongodb/src/db/comment.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  DBCommentAdapter,
+  CreateCommentArgs,
+  Comment,
+  GetCommentsArgs,
+  ConnectionResult,
+  CommentStatus
+} from '@wepublish/api'
+
+import {Collection, Db, FilterQuery} from 'mongodb'
+
+import {CollectionName, DBComment} from './schema'
+
+export class MongoDBCommentAdapter implements DBCommentAdapter {
+  private comments: Collection<DBComment>
+  private locale: string
+
+  constructor(db: Db, locale: string) {
+    this.comments = db.collection(CollectionName.Comments)
+    this.locale = locale
+  }
+
+  async createComment({input}: CreateCommentArgs): Promise<Comment> {
+    const {...data} = input
+    const {ops} = await this.comments.insertOne({
+      ...data,
+      revisions: [
+        {
+          ...data.revisions[0],
+          createdAt: new Date()
+        }
+      ],
+      createdAt: new Date(),
+      modifiedAt: new Date()
+    })
+
+    const {_id: id, ...comment} = ops[0]
+
+    return {id, ...comment}
+  }
+
+  async getComments({filter, limit}: GetCommentsArgs): Promise<ConnectionResult<Comment>> {
+    const metaFilters: FilterQuery<any> = []
+
+    if (filter?.status) {
+      metaFilters.push({status: filter.status})
+    }
+
+    const [totalCount, comments] = await Promise.all([
+      this.comments.countDocuments(), // MongoCountPreferences doesn't include collation
+
+      this.comments
+        .aggregate([], {collation: {locale: this.locale, strength: 2}})
+
+        .match(metaFilters.length ? {$and: metaFilters} : {})
+        .toArray()
+    ])
+
+    return {
+      nodes: comments.map<Comment>(({_id: id, ...comment}) => ({id, ...comment})),
+      pageInfo: {
+        startCursor: '',
+        endCursor: '',
+        hasNextPage: true,
+        hasPreviousPage: false
+      },
+      totalCount: totalCount
+    }
+  }
+
+  async getPublicComments(ids: readonly string[]): Promise<Comment[]> {
```

Comment:
Do you need that function? In which usecase do you want to get all approved comments? I think you need something ala `getCommentsForArticle` or `getCommentsForItem`.

Reply:
Why? I guess we need to have it for the editor users to get paginated lists so they can take action on the comment text, what's your opinion?

Reply:
We now use it in the context loader, which is being used for example in published article schema (Field resolver) to get its approved comments list.